### PR TITLE
rose bush: use os.stat to get last_activity_time in all rose bush pages.

### PR DESCRIPTION
**Pull request by arjclark**
Instead of performing an expensive query to get last activity time for rose bush pages (jobs/cycles) we can use the os.stat function instead as we do on the suites summary page.

@benfitzpatrick - please review 1
@oliver-sanders - please review 2

### CODE REVIEW

File: lib/python/rose/suite_engine_procs/cylc.py [hunk 1]
```diff
@@ -1021,10 +1021,10 @@ def get_suite_state_summary(self, user_name, suite_name):
         dao = self._db_init(self.SUITE_DB, user_name, suite_name)
         if not os.access(dao.db_f_name, os.F_OK | os.R_OK):
             return ret
-        stmt = "SELECT time FROM task_events ORDER BY time DESC LIMIT 1"
-        for row in self._db_exec(self.SUITE_DB, user_name, suite_name, stmt):
-            ret["last_activity_time"] = row[0]
-            break
+
+        ret["last_activity_time"] = strftime("%Y-%m-%dT%H:%M:%S+0000",
```
Comment (benfitzpatrick) [inline lib/python/rose/suite_engine_procs/cylc.py]: Can we have `Z` instead of `+0000`? (It's better ISO 8601, and cooler)

Comment (arjclark) [inline lib/python/rose/suite_engine_procs/cylc.py]: I can do. There's no requirement either way, that's just the way its already done on the existing suite listing page (and here), so I can change both lots accordingly. @matthewrmshin - are you happy for me to change that?

Comment (matthewrmshin) [inline lib/python/rose/suite_engine_procs/cylc.py]: Feel free. And perhaps feel free to combine the two bits of logic into one.

**Comment by arjclark** [comment]
@benfitzpatrick - Now using single function to get `last_activity_time` - popped in bush.py as discussed with @matthewrmshin. Now using `Z` for the timestamp of last activity and, for consistency, have also updated the other timestamps from `bush.py` to use `Z` rather than `+0000`.

**Comment by benfitzpatrick** [comment]
Looks good, tests as working and is faster - over to @oliver-sanders.

**Comment by oliver-sanders** [comment]
Tested as working. Looks OK.

**Pull request merged**